ignite/metrics/classification_report.py

```diff
@@ -0,0 +1,106 @@
+import json
+from typing import Callable, Collection, Dict, List, Optional, Union
+
+import torch
+
+from ignite.metrics.fbeta import Fbeta
+from ignite.metrics.metric import Metric
+from ignite.metrics.metrics_lambda import MetricsLambda
+from ignite.metrics.precision import Precision
+from ignite.metrics.recall import Recall
+
+__all__ = ["ClassificationReport"]
+
+
+def ClassificationReport(
+    beta: int = 1,
+    output_dict: bool = False,
+    output_transform: Callable = lambda x: x,
+    device: Union[str, torch.device] = torch.device("cpu"),
+    is_multilabel: bool = False,
+    labels: Optional[List[str]] = None,
+) -> MetricsLambda:
+    r"""Build a text report showing the main classification metrics. The report resembles in functionality to
+    `scikit-learn classification_report
+    <https://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html#sklearn.metrics.classification_report>`_
+    The underlying implementation doesn't use the sklearn function.
+
+    Args:
+        beta: weight of precision in harmonic mean
+        output_dict: If True, return output as dict, otherwise return a str
+        output_transform: a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+        is_multilabel: If True, the tensors are assumed to be multilabel.
+        device: optional device specification for internal storage.
+        labels: Optional list of label indices to include in the report
+
+    .. code-block:: python
+
+
+        y_true = torch.tensor([[1, 0],
+        [0, 1], [1, 0], [1, 1], [1, 0], [0, 1], [0, 1], [1, 0], [0, 1], [1, 0]])
+        y_pred = torch.randint(0, 2, size=(10,))
+
+        classification_report = ClassificationReport(output_dict=True)
+        classification_report.update((y_true, y_pred))
+        res = classification_report.compute()
```

Comment:
I've tried this example and I've got
> NotComputableError: Recall must have at least one example before it can be computed.

Reply:
@trsvchn  Thanks a lot for reviewing this.
The example was from the old implementation where I was overriding MetricsLambda and had a custom update method. This is not applicable anymore as MetricsLambda methods don't by default update underlying metrics.

The only way is to attach it to an engine as specified by the test cases. I will update the example.

Thanks for raising this!